src/crypto.cpp

```diff
@@ -352,9 +348,11 @@ int Crypto::SymEnc(
         CIPHER_TAG_SIZE,
         tag);
     if( res != 0 ) {
+        mbedtls_gcm_free(&ctx);
         spdlog::error("Failed to perform symmetric key encryption.");
         spdlog::error("Returned error: {}", to_string(res));
     }
+
```

Comment:
do we need to free the gcm context at the end of each function?

Reply:
If `mbedtls_gcm_init` or `mbedtls_gcm_setkey` allocates any memory than yes - but I'm not entirely sure if they do. I could look deeper if we want. Is there some reason we'd want to avoid freeing at the end of `SymEnc`/`SymDec`?

Reply:
no, no reason we'd want to avoid freeing. it seems that we call the `free` function if we error out before returning, e.g. [here](https://github.com/mc2-project/mc2-utils/pull/4/files/38b6108ac5a80e80cb7b0dc45236a20e34ef433d#diff-3700f1a6da19be70db791c8b323a328192eda62ce64f3f5b70c868e1bdbf63baR351). 

i think we should also free before the end of each function if we don't error

Reply:
oh woops good catch, thanks! I added the frees